librubyfmt/src/file_comments.rs

```diff
@@ -36,58 +36,48 @@ impl FileComments {
         fc
     }
 
+    /// Add a new comment. If the beginning of this file is a comment block,
+    /// each of those comment lines must be pushed before any other line, or
+    /// the end of the "start of file sled" will be incorrectly calculated.
     fn push_comment(&mut self, line_number: u64, l: String) {
-        if self.lowest_key == 0 {
-            self.lowest_key = line_number;
-        }
-
-        let last_line = self.contiguous_starting_indices.last();
-
-        let should_push =
-            line_number == 1 || (last_line.is_some() && last_line.unwrap() == &(line_number - 1));
-        if should_push {
-            self.contiguous_starting_indices.push(line_number);
+        match (&mut self.start_of_file_sled, line_number) {
+            (None, 1) => {
+                debug_assert!(
+                    self.other_comments.is_empty(),
+                    "If we have a start of file sled, it needs to come first,
+                     otherwise we won't know where the last line is",
+                );
+                self.start_of_file_sled = Some(CommentBlock::new(1..2, vec![l]));
+            }
+            (Some(sled), _) if sled.following_line_number() == line_number => {
+                sled.add_line(l);
```

Comment:
`l` looks like `1` here :\

Reply:
(I would pick a different variable name but I don't want to pointlessly churn the line declaring it. GH plz pick a less bad font)